specs/phase0/fork-choice.md

```diff
@@ -304,13 +362,57 @@ def store_target_checkpoint_state(store: Store, target: Checkpoint) -> None:
 
 ```python
 def update_latest_messages(store: Store, attesting_indices: Sequence[ValidatorIndex], attestation: Attestation) -> None:
-    target = attestation.data.target
+    attestation_slot = attestation.data.slot
     beacon_block_root = attestation.data.beacon_block_root
     for i in attesting_indices:
-        if i not in store.latest_messages or target.epoch > store.latest_messages[i].epoch:
-            store.latest_messages[i] = LatestMessage(epoch=target.epoch, root=beacon_block_root)
+        if i not in store.latest_messages or attestation_slot > store.latest_messages[i].slot:
+            node_key = get_node_key(beacon_block_root, attestation_slot)
+            store.latest_messages[i] = LatestMessage(slot=attestation_slot, root=node_key)
 ```
 
+#### `add_block_slot_node`
+
+```python
+def add_block_slot_node(store: Store, block: BeaconBlock, node_slot: Slot) -> BlockSlotNode:
```

Comment:
Is the returning `BlockSlotNode` for testing?

Reply:
That is for the recursion happening in the parent block's node check:
```python
store.block_slot_tree[parent_node] = add_block_slot_node(
                store, parent_block, parent_block_node_slot)
```